File: cases/taint/php/local-chain-negative/local_chain_negative.php
<?php
function dfb_source(): string { // DFB-SOURCE: local-chain-negative-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: local-chain-negative-sink

function run(): void {
    $first = dfb_source();
    $second = $first; // DFB-WITNESS: local-chain-negative-second
    $third = $second; // DFB-WITNESS: local-chain-negative-third
    dfb_sink("clean");
}

File: cases/taint/cpp/local-chain-negative/local_chain_negative.cpp
int dfb_source() { // DFB-SOURCE: local-chain-negative-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: local-chain-negative-sink

void run() {
    int first = dfb_source();
    int second = first; // DFB-WITNESS: local-chain-negative-second
    int third = second; // DFB-WITNESS: local-chain-negative-third
    dfb_sink(0);
}
